modules/data.remote/R/GEDI_AGB_prep.R

```diff
@@ -56,481 +92,437 @@ GEDI_AGB_prep <- function(site_info, time_points, outdir = file.path(getwd(), "G
-#' @param nfile.min the minimum required file number to be downloaded and extracted, default is 0.
-#' @param nrow.min the minimum required observation number to be extracted, default is 0.
-#' @param gradient the gradient for iteratively enlarge the extent if the nfile.min or nrow.min are not reached, default is 0. If nfile.min or nrow.min is 0 this will be skipped.
+#' Submit jobs through `qsub` for the `GEDI_L4A_Finder` function.
 #'
-#' @return A data frame containing AGB and sd for the target spatial and temporal extent.
+#' @title GEDI_L4A_Finder_batch
 #' 
-#' @examples
-#' @author Dongchen Zhang
-#' @importFrom magrittr %>%
-GEDI_AGB_download <- function(start_date, end_date, outdir, extent, nfile.min = 0, nrow.min = 0, gradient = 0) {
-  # download GEDI AGB files.
-  # if there is no data within current buffer distance.
-  files <- try(l4_download(ncore = 1,
-                           ul_lat = extent["ymax"], 
-                           lr_lat = extent["ymin"], 
-                           ul_lon = extent["xmin"], 
-                           lr_lon = extent["xmax"], 
-                           from = start_date, 
-                           to = end_date,
-                           outdir = outdir,
-                           just_path = T), silent = T)
-  # if we just need the data within fixed extent and hit error.
-  if ("try-error" %in% class(files) & nfile.min == 0) {
-    return(NA)
-  }
-  while ("try-error" %in% class(files) | length(files) < nfile.min) {
-    # we iteratively add 0.1 degree to the buffer distance.
-    extent[c(1, 4)] <- extent[c(1, 4)] + gradient
-    extent[c(2, 3)] <- extent[c(2, 3)] - gradient
-    files <- try(l4_download(ncore = 1,
-                             ul_lat = extent["ymax"], 
-                             lr_lat = extent["ymin"], 
-                             ul_lon = extent["xmin"], 
-                             lr_lon = extent["xmax"], 
-                             from = start_date, 
-                             to = end_date,
-                             outdir = outdir,
-                             just_path = T), silent = T)
+#' @param files Character: full paths of GEDI level 4A tiles.
+#' @param outdir Character: the physical path within which the batch job folders will be created.
+#' @param site_info List: list of site info including site_id, site_name, lon, and lat.
+#' @param num.folder Numeric: the number of batch folders to be created.
+#' @param buffer Numeric: buffer distance (in degree) that is used to create the bounding box (default is 0.005 [~ 500 m]).
+#' @param cores Numeric: numbers of core to be used for the parallel computation. The default is the maximum current CPU number.
+#' @param prerun Character: a vector of strings that will be executed beforehand. The default is NULL.
+#' 
+#' @return A list containing physical paths of GEDI tiles that intercept the each site.
+#' @export
+GEDI_L4A_Finder_batch <- function(files, 
```

Comment:
I'm not tracking closely, but as I scroll through this I'm seeing a lot of chunks that look very similar to the AGB code above. Are there more pieces that could be abstracted out into more general functions to reduce overall lines of code / make sure pieces that need to work in both contexts retain a shared implementation?

Reply:
I am thinking the same thing. But I would place it on the to-do list.